### DISCUSSION TITLE
Is there an option to check signed message from Solana Phantom wallet?

### DISCUSSION THREAD

**Discussion by mikield** [verze-app/solana-php-sdk#43] [Q&A]
Hello, all.

First of all I wanna say that I am beginner in crypto world, and I do not understand many things related to it.

But, I need to understand how to verify a signed message that was done by Phantom Wallet in my PHP project.

Could I do that somehow with this library?

**Comment by neverything** [comment]
There is a difference between `signMessage` and a signed transaction iirc. For logins, I'm using a react frontend that communicates with a backend written in Laravel/Livewire to handle the `signMessage` feature and validate logins and signups. To get you started, I highly recommend experimenting with the https://github.com/solana-labs/wallet-adapter/ to do the frontend part.

**Comment by mikield** [comment]
To all of you, who is looking "how to authenticate a user on Laravel backend" here I have an `AuthProvider` for you.

https://gist.github.com/mikield/47e38b03b0b4bafc04a19d5efc59d2b6

**Reply by neverything** [reply]
Thank you for sharing, one thing I added in verze was to check if the wallet actually does exist, but this might be an overkill depending on your application needs.

**Reply by mikield** [reply]
> Thank you for sharing, one thing I added in verze was to check if the wallet actually does exist, but this might be an overkill depending on your application needs.

How do you check the wallet exists? I mean, my wallet is not public visible to anyone in Solana system, cause I have not done any transaction using it.

So all you can check that it's a valid public key, and the message is signed by valid private key.

Also, if anyone want to modify how we retrieve user by their public key - these function should be changed.
![Знімок екрана 2022-09-05 о 07 17 16](https://user-images.githubusercontent.com/4073672/188365639-19204bb7-de55-47f6-9fa7-20763c7a3793.png)

**Reply by neverything** [reply]
You are absolutely correct, the wallets allowed to sign up on our end already have transactions or even assets like tokens and NFTs, so I'm checking on-chain if the accoun/wallet exists.
